tests/bud/no-history/Dockerfile

```diff
@@ -1,7 +1,8 @@
 # The important thing about that first base image is that it has no history
 # entries, but it does have at least one layer.
 
-FROM nixery.dev/shell AS first-stage
+FROM example.com/shell AS first-stage
```

Comment:
I was thinking anything on quay.io, or basically anything that does not introduce a new failure pain point.

Reply:
Hmm, using example.com has the advantage of producing an image reference that can't be confused with something that exists (or might once have existed, if we come back to this in the future) in an actual registry somewhere.  How strongly are you in favor of a quay.io name?

Reply:
I think my question was: instead of on-demand building each time, can this image be built once (with saved-committed-tracked script and instructions), say onto `quay.io/libpod/nohistory:20240422`, and then fetched on demand. This is a very difficult test to review. If any step ever breaks in a CI run, it will cost a lot of energy to investigate.

Reply:
I don't believe I'm authorized to push to that repository, so that's not a thing I can do in this PR, or set up for this PR to use.  The rest I can do.

Reply:
(Just in case it is of interest, https://datatracker.ietf.org/doc/html/rfc6761#section-6.4 specifies that the `.invalid` TLD should never resolve; that might be useful for similar situations.)